app/domain/authentication/authn_gcp/consts.rb

```diff
@@ -4,11 +4,22 @@ module Authentication
   module AuthnGcp
 
     PROVIDER_URI = "https://accounts.google.com"
-    PROJECT_ID_RESTRICTION_NAME = "project-id"
-    INSTANCE_NAME_RESTRICTION_NAME = "instance-name"
-    SERVICE_ACCOUNT_ID_RESTRICTION_NAME = "service-account-id"
-    SERVICE_ACCOUNT_EMAIL_RESTRICTION_NAME = "service-account-email"
-    PERMITTED_CONSTRAINTS = %w(instance-name project-id service-account-id service-account-email).freeze
+
+    module Restrictions
+
+      PROJECT_ID = "project-id"
+      INSTANCE_NAME = "instance-name"
+      SERVICE_ACCOUNT_ID = "service-account-id"
+      SERVICE_ACCOUNT_EMAIL = "service-account-email"
+
+      PERMITTED = [PROJECT_ID, INSTANCE_NAME, SERVICE_ACCOUNT_ID, SERVICE_ACCOUNT_EMAIL].freeze
+
+      CONSTRAINTS = Constraints::MultipleConstraint.new(
+        Constraints::AnyConstraint.new(any_of: PERMITTED),
```

Comment:
WDYM?
it creates a `AnyConstraint` with `PERMITTED` as its input.
It is created once per program lifetime.
Of course, if multiple authenticators will use it with their own parameters, it will be created once for each too.

Reply:
so AnyConstraint will look like `[PROJECT_ID, INSTANCE_NAME, SERVICE_ACCOUNT_ID, SERVICE_ACCOUNT_EMAIL]`

still do not understand what any_of does here 😄  please explain 🙏

Reply:
I added a comment on the class.
Is it clear now?